### PR TITLE
[Standalone for NativeComponents] RCTSlider

### PR DESCRIPTION
**Pull request by OdaDaisuke**
Changelog:

[iOS] [Changed] - Deal with #22990. Move `requireNativeComponent` to a separate file.

Test Plan:

- [x] yarn test
- [x] yarn flow
- [x] yarn lint

### CODE REVIEW

File: Libraries/Components/Slider/RCTSliderNativeComponent.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ *
+ * @format
+ * @flow strict-local
+ */
+
+'use strict';
+
+const requireNativeComponent = require('requireNativeComponent');
+
+module.exports = requireNativeComponent('RCTSlider');
```
[the remark this answers is not in the document]
Comment (OdaDaisuke) [reply Libraries/Components/Slider/RCTSliderNativeComponent.js]: I got it.

[the remark this answers is not in the document]
Comment (OdaDaisuke) [reply Libraries/Components/Slider/RCTSliderNativeComponent.js]: I got it.

Comment (TheSavior) [inline Libraries/Components/Slider/RCTSliderNativeComponent.js]: We want to have flow types for these components defined in these files too. Take a look at some of the other PRs as an example.

Comment (OdaDaisuke) [reply Libraries/Components/Slider/RCTSliderNativeComponent.js]: I see. I'll modify these components.

File: Libraries/Components/Slider/RCTSliderNativeComponent.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    maximumTrackTintColor?: ?ColorValue,
+    disabled?: ?boolean,
+    onValueChange?: ?(value: number) => void,
+    onSlidingComplete?: ?(value: number) => void,
+    testID?: ?string,
+|}>;
+
+type RCTSliderType = Class<NativeComponent<NativeProps>>;
+
+module.exports = ((requireNativeComponent('RCTSlider'): any): RCTSliderType;
```
[the remark this answers is not in the document]
Comment (TheSavior) [reply Libraries/Components/Slider/RCTSliderNativeComponent.js]: I think you are missing a closing paren here.

**Comment by OdaDaisuke** [comment]
Got it. I'll deal with it.

**Pull request closed without merging**